### PR TITLE
SUP 45978 script for restoring metadata profile and objects

### PR DESCRIPTION
**Pull request by Tova-a**

### CODE REVIEW

File: alpha/scripts/utils/restoreMetadataObjects.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
+<?php
+ini_set('memory_limit','1024M');
+if ($argc < 3)
```
Comment (drorsou) [inline alpha/scripts/utils/restoreMetadataObjects.php:3]:
```suggestion
if ($argc < 4)
```

File: alpha/scripts/utils/restoreMetadataObjects.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+$partnerId = $argv[1];
+$metadataProfileId = $argv[2];
+$metadataObjectIds = explode(',', $argv[3]);
+$totalMetadataObjectIds = count($metadataObjectIds);
+
+
+require("/opt/kaltura/app/alpha/scripts/bootstrap.php");
+
+$dryRun = true;
+if (isset($argv[3]) && $argv[3] == 'realrun')
```
Comment (drorsou) [inline alpha/scripts/utils/restoreMetadataObjects.php:19]: Argument realrun/dryrun is now at argv[4]

File: alpha/scripts/utils/restoreMetadataObjects.php [hunk 3]
```diff
@@ -0,0 +1,125 @@
+<?php
+ini_set('memory_limit','1024M');
+if ($argc < 3)
+{
+	echo PHP_EOL . ' ---- Restore Deleted Metadata Profile && Objects ---- ' . PHP_EOL;
+	echo ' Execute: php ' . $argv[0] . ' partnerId, metadata profile ID, [ /path/to/metadata object IDs list || metadataObjectId_1,metadataObjectId_2,.. || metadataObject ] [realrun / dryrun]' . PHP_EOL;
```
Comment (drorsou) [inline alpha/scripts/utils/restoreMetadataObjects.php:6]: Do you handle path to metadata?
Or only the list of ids?

File: alpha/scripts/utils/restoreMetadataObjects.php [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+	{
+		$metadataObjectId = trim($metadataObjectId);
+		KalturaLog::debug("=== Metadata Object Id: " . $metadataObjectId);
+		MetadataPeer::setUseCriteriaFilter(false);
+		$metadataObject = MetadataPeer::retrieveByPK($metadataObjectId);
+		MetadataPeer::setUseCriteriaFilter(true);
+
+		/* @var $metadataObjectId metadata */
+
+		if (!$metadataObjectId)
```
Comment (drorsou) [inline alpha/scripts/utils/restoreMetadataObjects.php:73]:
```suggestion
		if (!$metadataObject)
```

Comment (drorsou) [inline alpha/scripts/utils/restoreMetadataObjects.php:73]: Should verify the metadata object requires restoring, check its status before continue

File: alpha/scripts/utils/restoreMetadataObjects.php [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+		{
+			KalturaLog::debug("ERR3 - metadata object id " . $metadataObjectId . " not found");
+			continue;
+		}
+
+		KalturaLog::debug('=== Restoring metadata object id [' . $metadataObject->getID() . ']');
+		$mdOProfileVersion = $metadataObject->getMetadataProfileVersion();
+
+		if ($mdOProfileVersion <> $mdVersion)
```
Comment (drorsou) [inline alpha/scripts/utils/restoreMetadataObjects.php:82]: I would put this after checking the metadataObject status.
No need to print errors for versions difference, if the status isn't deleted

File: alpha/scripts/utils/restoreMetadataObjects.php (hunk 3 above)
Comment (Tova-a) [reply alpha/scripts/utils/restoreMetadataObjects.php:6]: Added handlig to path.

File: alpha/scripts/utils/restoreMetadataObjects.php (hunk 2 above)
Comment (Tova-a) [reply alpha/scripts/utils/restoreMetadataObjects.php:19]: changed.

File: alpha/scripts/utils/restoreMetadataObjects.php (hunk 1 above)
Comment (Tova-a) [reply alpha/scripts/utils/restoreMetadataObjects.php:3]: changed.

File: alpha/scripts/utils/restoreMetadataObjects.php (hunk 4 above)
Comment (Tova-a) [reply alpha/scripts/utils/restoreMetadataObjects.php:73]: changed.

**Review by drorsou — APPROVED** [review]

**Pull request merged**